streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -200,6 +201,30 @@ public RocksDBMetricsRecordingTrigger rocksDBMetricsRecordingTrigger() {
         }
     }
 
+    public <T> void addThreadLevelImmutableMetric(final String name,
+        final String description,
+        final String threadId,
+        final T value) {
+        final MetricName metricName = metrics.metricName(
+            name, THREAD_LEVEL_GROUP, description, threadLevelTagMap(threadId));
+        synchronized (threadLevelMetrics) {
```

Comment:
Why we want to not prefix the thread id but directly use it as the key in the `threadLevelMetrics` map? Ditto below.

Reply:
not sure I follow the question here - we are using `threadLevelMetrics` to track the metrics for each thread so they can be cleaned up later on when the thread exits. What's wrong with using the thread id for that?

Reply:
Oh it's totally fine to use the thread id, it's just that for other thread-level metrics we would prefix the thread id with either `internal` or `external`, i.e. the "key" would not just be the thread id itself.

Reply:
Ah I wasn't aware of the external vs internal sensor names. Now that I read through this again it seems to be that some external caller with a reference to `StreamsMetrics` can add their own sensors, which don't get cleaned up when the thread goes away. In this case we don't have external callers adding any thread-level metrics to the map, so we don't really need the prefix. Happy to include it to keep things consistent.

Reply:
Yes, I'm only suggesting this as to keep things consistent, not really a correctness issue.

As for `external` metrics, when users create their own metrics I think they are responsible for e.g. de-registering them upon closure, not Streams itself.